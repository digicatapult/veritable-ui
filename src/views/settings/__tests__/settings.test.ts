import { expect } from 'chai'
import { describe, it } from 'mocha'
import SettingsTemplates from '../settings.js'

describe('SettingsTemplates', () => {
  describe('settings', () => {
    it('should render settings page', async () => {
      const templates = new SettingsTemplates()
      const rendered = await templates.settings({
        company_name: 'Test Company Name',
        companies_house_number: '0000000',
        from_email: 'mail@testmail.com',
        postal_address: 'Some Address, Somewhere',
        admin_email: 'admin@testmail.com',
      })
      expect(rendered).to.matchSnapshot()
    })
  })

  // should I add a test for escaping html in name?
})
