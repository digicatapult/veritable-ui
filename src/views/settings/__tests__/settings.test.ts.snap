// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsTemplates settings should render settings page 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon-dark.ico\\" media=\\"(prefers-color-scheme: dark)\\"/><link rel=\\"icon\\" type=\\"image/ico\\" href=\\"/public/images/favicon.ico\\" media=\\"(prefers-color-scheme: light), (prefers-color-scheme: no-preference)\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/settings.css\\"/><title>Veritable - Settings</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><div class=\\"profile-icon\\" style=\\"background-color:#38b6ff;\\">A</div><a title=\\"categories\\" href=\\"/\\" data-active style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"queries\\" href=\\"/queries\\" style=\\"--background-image:url('/public/images/query.svg');\\"></a><a title=\\"credentials\\" href=\\"/credentials\\" style=\\"--background-image:url('/public/images/credential.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"settings\\" href=\\"/settings\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>Settings</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Settings\\" href=\\"/settings\\">Settings</a></div></div><div id=\\"content-main\\"><div id=\\"settings-container\\"><h1>Settings</h1><h2><i>Account Details</i></h2><form id=\\"settings-form\\" hx-post=\\"/settings/update\\" hx-swap=\\"innerHTML\\"><label for=\\"company_name\\" class=\\"settings-input-label\\">Company Name</label><input id=\\"company_name\\" class=\\"disabled settings-input\\" name=\\"company_name\\" placeholder=\\" placeholder DIGITAL CATAPULT\\" required value=\\"Test Company Name\\" type=\\"text\\" oninput=\\"this.reportValidity()\\" minlength=\\"1\\"/><label for=\\"companies_house_number\\" class=\\"settings-input-label\\">Companies House Number</label><input id=\\"companies_house_number\\" class=\\"disabled settings-input\\" name=\\"companies_house_number\\" required value=\\"0000000\\" type=\\"text\\" oninput=\\"this.reportValidity()\\" minlength=\\"1\\"/><label for=\\"postal_address\\" class=\\"settings-input-label\\">Postal Address</label><input id=\\"postal_address\\" class=\\"disabled settings-input\\" name=\\"postal_address\\" required value=\\"Some Address, Somewhere\\" type=\\"text\\" oninput=\\"this.reportValidity()\\" minlength=\\"1\\"/><label for=\\"from_email\\" class=\\"settings-input-label\\">From Address</label><input id=\\"from_email\\" class=\\"disabled settings-input\\" name=\\"from_email\\" required value=\\"mail@testmail.com\\" type=\\"text\\" oninput=\\"this.reportValidity()\\" minlength=\\"1\\"/><label for=\\"admin_email\\" class=\\"settings-input-label\\">Administrator Email</label><div class=\\"in-row\\"><input id=\\"admin_email\\" class=\\"settings-input extra-padding-right\\" name=\\"admin_email\\" required value=\\"admin@testmail.com\\" type=\\"text\\" oninput=\\"this.reportValidity()\\" minlength=\\"1\\" readonly/><input type=\\"image\\" src=\\"/public/images/edit.svg\\" class=\\"edit-button\\" name=\\"action\\" value=\\"updateSettings\\"/></div></form><div id=\\"reset-container\\"><h2>Reset (Demo only)</h2><p>Reset or delete all current connections. This will remove any pending, active or other connections associated with your account.</p><button id=\\"reset-btn\\" hx-delete=\\"/reset\\" hx-target=\\"this\\" hx-swap=\\"none\\" style=\\"filled\\"><img id=\\"reset-image\\" src=\\"/public/images/reset.svg\\" alt=\\"Reset\\"/>Reset</button></div></div></div></main></body></html>"`;
